cumulusci/core/flowrunner.py

```diff
@@ -465,3 +488,17 @@ def _init_org(self):
         if self.org_config.config != orig_config:
             self.logger.info("Org info has changed, updating org in keychain")
             self.project_config.keychain.set_org(self.org_config)
+
+    def resolve_return_value_options(self, options):
+        """Handle dynamic option value lookups in the format ^^task_name.attr"""
+        for key, value in options.items():
+            if isinstance(value, str) and value.startswith(RETURN_VALUE_OPTION_PREFIX):
+                path, name = value[len(RETURN_VALUE_OPTION_PREFIX) :].rsplit(".", 1)
```

Comment:
not for this PR, but i've longed to write an actual grammar for options values that handles ^^ and $ more centrally

Reply:
It'd also be nice to catch invalid paths during init rather than during run, but it looks nontrivial (needs a second validation pass after the steps have been initted I think) and isn't a regression, so let's not do it now.